ci/test/lint-deps.toml

```diff
@@ -0,0 +1,68 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Configuration file for cargo-deplint
+#
+# If you find yourself here because CI failed on a PR, please don't blindly edit
+# out lints :D. Danhhz is always happy to discuss crate structure if you need
+# help fitting a change into these rules.
+
+# Keep various large crates from depending on frequently changing ones, so that
+# development iteration doesn't invalidate them.
+[[deny]]
+name = "mz-transform"
+dependencies = [
+    "mz-compute-client",
+    "mz-storage-client",
+]
+
+[[deny]]
+name = "mz-sql"
+dependencies = [
+    "mz-compute-client",
+    "mz-storage-client",
+]
+
+[[deny]]
+name = "mz-compute"
+dependencies = [
+    # TODO: "mz-storage-controller",
+    "mz-storage",
+]
+
+[[deny]]
+name = "mz-storage"
+dependencies = [
+    # TODO: "mz-storage-controller",
+    "mz-compute",
+]
+
+# Keep various crates in only one of environmentd or clusterd.
+[[deny]]
+name = "environmentd"
+dependencies = [
+    "mz-compute",
+    "mz-storage",
+    "mz-storage-operators",
+]
```

Comment:
Could we add an entry for `rocksdb`? We want to make sure it doesn't creep into `environmentd` again.

Reply:
Great idea! Done